src/main/java/io/fabric8/maven/docker/config/CopyConfiguration.java

```diff
@@ -0,0 +1,142 @@
+         * same name will be created in the hostDirectory, i.e. not just content of directory is copied, but the same
+         * name directory is created too.
+         */
+        private String hostDirectory;
+
+        public Entry() {}
+
+        public Entry(String containerPath, String hostDirectory) {
+            this.containerPath = containerPath;
+            this.hostDirectory = hostDirectory;
+        }
+
+        public String getContainerPath() {
+            return containerPath;
+        }
+
+        public String getHostDirectory() {
+            return hostDirectory;
+        }
+
+        public void setContainerPath(String containerPath) {
+            this.containerPath = containerPath;
+        }
+
+        public void setHostDirectory(String hostDirectory) {
+            this.hostDirectory = hostDirectory;
+        }
+    }
+
+    /**
+     * Items to copy from container.
+     */
+    @Parameter
+    private List<Entry> entries;
+
+    public List<Entry> getEntries() {
+        return entries;
+    }
+
+    public List<Properties> getEntriesAsListOfProperties() {
+        if (entries == null) {
+            return null;
+        }
+        final List<Properties> properties = new ArrayList<>(entries.size());
+        for (Entry entry : entries) {
+            final Properties entryProperties = new Properties();
+            entryProperties.put(CONTAINER_PATH_PROPERTY, entry.getContainerPath());
+            final String hostDirectory = entry.getHostDirectory();
+            if (hostDirectory != null) {
+                entryProperties.put(HOST_DIRECTORY_PROPERTY, hostDirectory);
+            }
+            properties.add(entryProperties);
+        }
+        return properties;
+    }
+
+    public static class Builder {
+
+        private final CopyConfiguration config;
+
+        public Builder() {
+            this(null);
+        }
+
+        public Builder(CopyConfiguration that) {
+            if (that == null) {
+                config = new CopyConfiguration();
+            } else {
+                config = DeepCopy.copy(that);
+            }
+        }
+
+        public Builder entries(List<Entry> entries) {
+            final List<Entry> entriesCopy = new ArrayList<>(entries.size());
+            for (Entry entry : entries) {
+                entriesCopy.add(new Entry(entry.getContainerPath(), entry.getHostDirectory()));
+            }
+            config.entries = entriesCopy;
+            return this;
+        }
+
+        public Builder entriesAsListOfProperties(List<Properties> entries) {
+            config.entries = getEntriesFromProperties(entries);
+            return this;
+        }
+
+        public CopyConfiguration build() {
+            return config;
+        }
+
+        private static List<Entry> getEntriesFromProperties(List<Properties> properties) {
+            if (properties == null) {
+                return null;
```

Comment:
```suggestion
                return Collections.emptyList();
```

Reply:
Refer to `io.fabric8.maven.docker.config.RunImageConfiguration#getWaitConfiguration` method. I expect that `io.fabric8.maven.docker.config.CopyConfiguration#entries` field should be `null` to follow existing design.